src/Quizzer.API.BL/Facades/QuestionFacade.cs

```diff
@@ -0,0 +1,310 @@
+                if (answer is null) {
+                    SelectedAnswerEntity newAnswer = NewAnswer(answerEntity!.Id, userAnswer.Id);
+
+                    selectedAnswerRepository.Insert(newAnswer);
+                } else {
+                    selectedAnswerRepository.Delete(answer.Id);
+                }
+            }
+
+            void HandleOrderChoice() {
+                SelectedAnswerEntity? answer = userAnswer.SelectedAnswers.SingleOrDefault(
+                  s => s.AnswerId == answerEntity!.Id);
+
+                IEnumerable<SelectedAnswerEntity> otherAnswersWithSameOrder = userAnswer.SelectedAnswers
+                    .Where(a => a.AnswerId != answerId && a.Order == order);
+
+                foreach (SelectedAnswerEntity otherAnswer in otherAnswersWithSameOrder) {
+                    otherAnswer.Order = 0;
+                    selectedAnswerRepository.Update(otherAnswer);
+                }
+
+                if (answer is null) {
+                    if (order == 0) throw new ArgumentException("Cannot create ordered answer without order");
+
+                    answer = NewAnswer(answerEntity!.Id, userAnswer.Id);
+
+                    selectedAnswerRepository.Insert(answer);
+                }
+
+                if (order == 0) {
+                    selectedAnswerRepository.Delete(answer.Id);
+                } else {
+                    answer.Order = order;
+                    selectedAnswerRepository.Update(answer);
+                }
+            }
+
+            #endregion
+        }
+
+        private void HandleFormatChange(QuestionEntity entity, AnswerFormat currentFormat) {
+            if (entity.Type == currentFormat) return;
+
+            IUnitOfWork uow = p_unitOfWorkFactory.Create();
+
+            using IRepository<UserAnswerEntity> userAnswerRepository = uow.GetRepository<UserAnswerEntity>();
+
+            IEnumerable<UserAnswerEntity> userAnswers = userAnswerRepository.GetByPredicate(
+                e => e.QuestionId == entity.Id, nameof(UserAnswerEntity.Question));
+
+            foreach (UserAnswerEntity userAnswer in userAnswers) {
+                userAnswerRepository.Delete(userAnswer.Id);
+            }
+        }
+
+        private static SelectedAnswerEntity NewAnswer(Guid answerId, Guid userAnswerId) {
+            return new SelectedAnswerEntity() {
+                Id = Guid.NewGuid(),
+                AnswerId = answerId,
+                Order = 0,
+                UserAnswerId = userAnswerId
+            };
+        }
+
+        private static void CheckQuizTimeFrame(DateTime answerTime, AnswerEntity answerEntity) {
+            QuizEntity quiz = answerEntity.Question!.Quiz!;
+
+            // Answering in invalid time frame
+            if (answerTime < quiz.StartTime || answerTime > quiz.EndTime) {
+                throw new ArgumentException("Quiz is closed");
+            }
+        }
+
+        private static AnswerEntity GetValidAnswer(Guid answerId, IUnitOfWork uow) {
```

Comment:
What are use cases of this function?

Reply:
Returns valid answer based on answer Id with mapped corresponding question and quiz to the answer. For invalid Id throws error. This is internal function used to decouple logic blocks into the methods.